Sources/SwiftSyntaxMacros/MacroSystem.swift

```diff
@@ -438,6 +439,23 @@ extension MacroApplication {
       }
     }
 
+    let extensionMacroAttrs = getMacroAttributes(attachedTo: decl.as(DeclSyntax.self)!, ofType: ExtensionMacro.Type.self)
+    let extendedTypeSyntax = TypeSyntax("\(extendedType.trimmed)")
```

Comment:
In the [proposal](https://github.com/hborla/swift-evolution/blob/extension-macros/proposals/NNNN-extension-macros.md#implementing-extension-macros) it's mentioned: 

> Each `ExtensionDeclSyntax` in the resulting array must use the `providingExtensionsOf` parameter as the extended > type, which is a qualified type name. For example, for the following code:
> 
> ```swift
> struct Outer {
>   @MyProtocol
>   struct Inner {}
> }
> ```
> The type syntax passed to `ExtensionMacro.expansion` for `providingExtensionsOf` is `Outer.Inner`.

I'm not sure if I've understood this part of the proposal correctly. Does it mean that if I create this macro:
```swift
public struct SomeExtensionMacro: ExtensionMacro {
  public static func expansion(
    of node: AttributeSyntax,
    attachedTo declaration: some DeclGroupSyntax,
    providingExtensionsOf type: some TypeSyntaxProtocol,
    in context: some MacroExpansionContext
  ) throws -> [ExtensionDeclSyntax] {
    [ExtensionDeclSyntax(extendedType: type, memberBlock: MemberDeclBlockSyntax {})]
  }
}
```
and I attach it like so
```swift
struct Outer {
  @SomeExtensionMacro
  struct Inner {}
}
```
are we expecting that `extendedType.description` here will look something like this: `Outer.Inner`? If I understand the assumptions correctly, I think this would happen because the code above:
```swift
let extendedType: Syntax
if let identified = decl.asProtocol(IdentifiedDeclSyntax.self) {
  extendedType = Syntax(identified.identifier.trimmed)
} else if let ext = decl.as(ExtensionDeclSyntax.self) {
  extendedType = Syntax(ext.extendedType.trimmed)
} else {
  return []
}
```
can only extract the identifier from the root type.

Here is a test case covering this example using `@AddSendableExtension` macro:
```swift
func testExtensionExpansionOfNestedType() {
  assertMacroExpansion(
    """
    struct Outer {
      @AddSendableExtension
      struct Inner {
      }
    }
    """,
    expandedSource:
      """

      struct Outer {
        struct Inner {
        }
      }
      extension Outer.Inner: Sendable {
      }
      """,
    macros: testMacros,
    indentationWidth: indentationWidth
  )
}
```
right now it's failing with message: 
```
failed - Actual output (+) differed from expected output (-):
 struct Outer {
   struct Inner {
   }
 }
–extension Outer.Inner: Sendable {
–}

Actual expanded source:

struct Outer {
  struct Inner {
  }
}
extension Outer.Inner: Sendable {
}
```
Am I understanding it correctly that this shouldn't fail?

Reply:
This is a great question, and there's a little bit of nuance here because the expansion performed by the `MacroSystem` in SwiftSyntax is only used for unit testing macros, but the real expansion is integrated into the compiler (and implemented in https://github.com/apple/swift/pull/66967). It's not possible for `MacroSystem` to form the same fully qualified type that the compiler can form in the general case, e.g. the compiler can look through type aliases, while the macro system here cannot. I believe there are also some known issues with where the expansion is inserted in `MacroSystem` for conformance macros, because it doesn't insert them at the top-level like the compiler does. I definitely think we can improve what `MacroSystem` does in many cases (e.g. we could form a qualified name from the lexical nesting structure), but it can't perfectly match what the compiler does during extension macro expansion.